velox/exec/Window.cpp

```diff
@@ -379,25 +450,71 @@ void Window::callApplyForPartitionRows(
         }
         break;
       }
-      case core::WindowNode::BoundType::kPreceding:
-      case core::WindowNode::BoundType::kFollowing:
-        VELOX_NYI("Not supported");
+      case core::WindowNode::BoundType::kPreceding: {
+        kBoundFrameCheck(frameArg);
```

Comment:
This shouldn't be checked each time. I think we can assume that it was correctly set by the code.